### DISCUSSION TITLE
Ideas on how to tackle basic NULL pointer exceptions

### DISCUSSION THREAD

**Discussion by thealberto** [libpng/libpng#7] [General]
Hi all,
so I have spent some time reading the source and I have noticed that a number of functions which are exported by the `so` could cause `NULL pointer exception`. An example is the `png_save_int_32` function which has the following source code at the moment

```
void PNGAPI
png_save_int_32(png_bytep buf, png_int_32 i)
{
   png_save_uint_32(buf, (png_uint_32)i);
}
```

So far I have identified `9` cases where with more or less simple code I could force `libpng` to crash. Would you consider worth to spend some time trying to identity ( ideally ) all the basic situations  and fix them all?

If so, do you want a single PR which will take care of them or one PR for each fix so we can always revert it back if needed?

Thanks

**Comment by ioquatix** [comment]
Yes, I would love that.

**Comment by thealberto** [comment]
Hey, I'll continue to work against but I think i'll try to have this work directly merge into the main repo. What do you think?

**Reply by ioquatix** [reply]
I don't mind what you do.
